// @share-package/common/build/events/subjects.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Subjects = void 0;
var Subjects;
(function (Subjects) {
    Subjects["TicketCreated"] = "ticket:created";
    Subjects["TicketUpdated"] = "ticket:updated";
    Subjects["OrderCreated"] = "order:created";
    Subjects["OrderCancelled"] = "order:cancelled";
    Subjects["ExpirationComplete"] = "expiration:complete";
    // User status
    Subjects["UserCreated"] = "user:created";
    Subjects["UserUpdated"] = "user:updated";
    Subjects["UserCancelled"] = "user:canceller";
    // Role status
    Subjects["RoleCreated"] = "role:created";
    Subjects["RoleUpdated"] = "role:updated";
    Subjects["RoleDeleted"] = "role:delete";
    // Permission
    Subjects["PermissionCreated"] = "permission:created";
    Subjects["PermissionUpdated"] = "permission:updated";
    Subjects["PermissionDeleted"] = "permission:delete";
    // Role-Permission
    Subjects["RolePermissionCreated"] = "role-permission:created";
    //UserURMapping
    Subjects["UserURMCreated"] = "user-urm:created";
    Subjects["UserURMDeleted"] = "user-urm:delete";
})(Subjects || (exports.Subjects = Subjects = {}));
