// @share-package/common/build/events/types/permissions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerPermission = exports.ListPermission = void 0;
var ListPermission;
(function (ListPermission) {
    // Create, read, update, delete orders
    ListPermission["OrderCreate"] = "order.create";
    ListPermission["OrdersRead"] = "order.read";
    ListPermission["OrderUpdate"] = "order.update";
    ListPermission["OrderDelete"] = "order.delete";
    // Create, read, update, delete shifts
    ListPermission["ShiftCreate"] = "shift.create";
    ListPermission["ShiftRead"] = "shift.read";
    ListPermission["ShiftUpdate"] = "shift.update";
    ListPermission["ShiftDelete"] = "shift.delete";
    // Create, read, update, delete employeee
    ListPermission["EmployeeCreate"] = "employee.create";
    ListPermission["EmployeeRead"] = "employee.read";
    ListPermission["EmployeeUpdate"] = "employee.update";
    ListPermission["EmployeeDelete"] = "employee.delete";
    // Create, read, update, delete customers
    ListPermission["CustomerCreate"] = "customer.create";
    ListPermission["CustomerRead"] = "customer.read";
    ListPermission["CustomerUpdate"] = "customer.update";
    ListPermission["CustomerDelete"] = "customer.delete";
    // Create, read, update, delete permissions
    // AdminPermissions = 'admin.permissions',
    // ReadAndWritePermission = 'create.and.write.permission',
    // ReadDatabasePermisson = 'read.database.permission',
    // Create, read, update, delete roles
    ListPermission["RoleCreate"] = "role.create";
    ListPermission["RoleRead"] = "role.read";
    ListPermission["RoleUpdate"] = "role.update";
    ListPermission["RoleDelete"] = "role.deleted";
    // Create, read, update, delete services
    ListPermission["ServiceCreate"] = "service.create";
    ListPermission["ServiceRead"] = "service.read";
    ListPermission["ServiceUpdate"] = "service.update";
    ListPermission["ServiceDelete"] = "service.deleted";
    // Create, read, update, delete package service
    ListPermission["PackageCreate"] = "package.create";
    ListPermission["PackageRead"] = "package.read";
    ListPermission["PackageUpdate"] = "package.update";
    ListPermission["PackageDelete"] = "package.deleted";
})(ListPermission || (exports.ListPermission = ListPermission = {}));
var CustomerPermission;
(function (CustomerPermission) {
    CustomerPermission["OrderCreate"] = "order.create";
    CustomerPermission["OrdersRead"] = "order.read";
    CustomerPermission["OrderUpdate"] = "order.update";
    CustomerPermission["OrderDelete"] = "order.delete";
})(CustomerPermission || (exports.CustomerPermission = CustomerPermission = {}));
