// @share-package/common/build/events/types/permissions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListPermission = void 0;
var ListPermission;
(function (ListPermission) {
    // Create, read, update, delete orders
    ListPermission["OrderCreate"] = "order.create";
    ListPermission["OrdersRead"] = "order.read";
    ListPermission["OrderUpdate"] = "order.update";
    ListPermission["OrderDelete"] = "order.delete";
    // Create, read, update, delete shifts
    ListPermission["ShiftCreate"] = "shift.create";
    ListPermission["ShiftRead"] = "shift.read";
    ListPermission["ShiftUpdate"] = "shift.update";
    ListPermission["ShiftDelete"] = "shift.delete";
    // Create, read, update, delete employeee
    ListPermission["EmployeeCreate"] = "employee.create";
    ListPermission["EmployeeRead"] = "employee.read";
    ListPermission["EmployeeUpdate"] = "employee.update";
    ListPermission["EmployeeDelete"] = "employee.delete";
    // Create, read, update, delete customers
    ListPermission["CustomerCreate"] = "customer.create";
    ListPermission["CustomerRead"] = "customer.read";
    ListPermission["CustomerUpdate"] = "customer.update";
    ListPermission["CustomerDelete"] = "customer.delete";
    // Create, read, update, delete permissions
    // AdminPermissions = 'admin.permissions',
    // ReadAndWritePermission = 'create.and.write.permission',
    // ReadDatabasePermisson = 'read.database.permission',
    // Create, read, update, delete roles
    ListPermission["RoleCreate"] = "create.role";
    ListPermission["RoleRead"] = "read.role";
    ListPermission["RoleUpdate"] = "update.role";
    ListPermission["RoleDelete"] = "deleted.role";
})(ListPermission || (exports.ListPermission = ListPermission = {}));
