// @share-package/common/build/events/types/user-permission.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagerPermission = exports.EmployeePermission = exports.CustomerPermission = void 0;
var CustomerPermission;
(function (CustomerPermission) {
    CustomerPermission["CreateOrder"] = "create-order";
    CustomerPermission["ViewOwnerOrders"] = "view-orders";
    CustomerPermission["CancelledOrders"] = "cancelled-orders";
    CustomerPermission["ChangeProfile"] = "change-profile";
})(CustomerPermission || (exports.CustomerPermission = CustomerPermission = {}));
var EmployeePermission;
(function (EmployeePermission) {
    EmployeePermission["ViewOrdersOnDay"] = "view-orders";
    EmployeePermission["CreateOrder"] = "create-order";
    EmployeePermission["CancelOrdersOnDay"] = "view-orders-on-day";
    EmployeePermission["ChangeProfile"] = "change-profile";
    // View your own shift
    EmployeePermission["ViewShift"] = "view-shift";
})(EmployeePermission || (exports.EmployeePermission = EmployeePermission = {}));
var ManagerPermission;
(function (ManagerPermission) {
    // Create, read, update, delete orders
    ManagerPermission["CRUDOrders"] = "crud-orders";
    ManagerPermission["ReadAndWriteOrder"] = "create-and-write-orders";
    ManagerPermission["ReadDatabaseOrder"] = "read-database-order";
    // Create, read, update, delete shifts
    ManagerPermission["CRUDShifts"] = "crud-shifts";
    ManagerPermission["ReadAndWriteShift"] = "create-and-write-shift";
    ManagerPermission["ReadDatabaseShift"] = "read-database-shift";
    // Create, read, update, delete employeee
    ManagerPermission["CRUDEmployees"] = "crud-employees";
    ManagerPermission["ReadAndWriteEmpolyee"] = "create-and-write-employee";
    ManagerPermission["ReadDatabaseEmployee"] = "read-database-employee";
    // Create, read, update, delete customers
    ManagerPermission["CRUDCustomers"] = "crud-customers";
    ManagerPermission["ReadAndWriteCustomer"] = "create-and-write-customer";
    ManagerPermission["ReadDatabaseCustomer"] = "read-database-customer";
    // Create, read, update, delete permissions
    ManagerPermission["CRUDPermissions"] = "crud-permissions";
    ManagerPermission["ReadAndWritePermission"] = "create-and-write-permission";
    ManagerPermission["ReadDatabasePermisson"] = "read-database-permission";
    // Create, read, update, delete roles
    ManagerPermission["CRUDRoles"] = "crud-roles";
    ManagerPermission["ReadAndWriteRole"] = "create-and-write-employee";
    ManagerPermission["ReadDatabaseRole"] = "read-database-employee";
})(ManagerPermission || (exports.ManagerPermission = ManagerPermission = {}));
