// @share-package/common/build/events/types/user-permission.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagerPermission = exports.EmployeePermission = exports.CustomerPermission = void 0;
var CustomerPermission;
(function (CustomerPermission) {
    CustomerPermission["Order"] = "order";
    CustomerPermission["ViewOwnerOrders"] = "view-orders";
    CustomerPermission["CancelledOrders"] = "cancelled-orders";
    CustomerPermission["ChangeProfile"] = "change-profile";
})(CustomerPermission || (exports.CustomerPermission = CustomerPermission = {}));
var EmployeePermission;
(function (EmployeePermission) {
    EmployeePermission["ViewOrdersOnDay"] = "view-orders";
    EmployeePermission["CancelOrdersOnDay"] = "view-orders-on-day";
    EmployeePermission["ChangeProfile"] = "change-profile";
    // View your own shift
    EmployeePermission["ViewShift"] = "view-shift";
})(EmployeePermission || (exports.EmployeePermission = EmployeePermission = {}));
var ManagerPermission;
(function (ManagerPermission) {
    // Create, read, update, delete orders
    ManagerPermission["CRUDOrders"] = "crud-orders";
    // Create, read, update, delete shifts
    ManagerPermission["CRUDShifts"] = "crud-shifts";
    // Create, read, update, delete employeee
    ManagerPermission["CRUDEmployees"] = "crud-employees";
    // Create, read, update, delete customers
    ManagerPermission["CRUDCustomers"] = "crud-customers";
})(ManagerPermission || (exports.ManagerPermission = ManagerPermission = {}));
