// @share-package/common/build/events/subjects.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Subjects = void 0;
var Subjects;
(function (Subjects) {
    Subjects["TicketCreated"] = "ticket:created";
    Subjects["TicketUpdated"] = "ticket:updated";
    Subjects["OrderCreated"] = "order:created";
    Subjects["OrderCancelled"] = "order:cancelled";
    Subjects["ExpirationComplete"] = "expiration:complete";
    //--------------- User status-----------------
    Subjects["UserCreated"] = "user:created";
    Subjects["UserUpdated"] = "user:updated";
    Subjects["UserDeleted"] = "user:deleted";
    //--------------- Role status---------------
    Subjects["RoleCreated"] = "role:created";
    Subjects["RoleUpdated"] = "role:updated";
    Subjects["RoleDeleted"] = "role:delete";
    //--------------- Permission--------------
    Subjects["PermissionCreated"] = "permission:created";
    Subjects["PermissionUpdated"] = "permission:updated";
    Subjects["PermissionDeleted"] = "permission:delete";
    //--------------- Role-Permission----------------
    Subjects["RolePermissionCreated"] = "role-permission:created";
    Subjects["RolePermissionDeleted"] = "role-permission:deleted";
    //---------------UserURMapping-------------------
    Subjects["UserURMCreated"] = "user-urm:created";
    Subjects["UserURMDeleted"] = "user-urm:delete";
    // ---------------product----------------------------
    Subjects["ProductCreated"] = "product:created";
    Subjects["ProductUpdated"] = "product:updated";
    Subjects["ProductDeleted"] = "product:deleted";
})(Subjects || (exports.Subjects = Subjects = {}));
